.github/workflows/shippable_builds.yml

```diff
@@ -792,6 +828,68 @@ jobs:
           releaseFiles: |
             uploads/${{ steps.rename.outputs.PKG_FILE }}
 
+      - name: Determine FTP Destination
+        id: ftp_destination
+        if: ${{ !inputs.skipFtp && contains(matrix.releaseTarget, 'ftp') && matrix.packageFormat == 'apk' }}
+        shell: bash
+        env:
+          RELEASE_TYPE: ${{ env.RELEASE_TYPE }}
+          VERSION_NAME: ${{ steps.pkginfo.outputs.VERSION_NAME }}
+          UPLOAD_FTP_STAGE: ${{ inputs.uploadFtpStage }}
+        run: |
+          GCP_ENV=prod
+          GCP_PROJECT=moz-fx-productdelivery-pr-38b5
+          FTP_URL_PREFIX="https://ftp.mozilla.org/pub/thunderbird-mobile/android/"
+          if [[ "$UPLOAD_FTP_STAGE" = "true" ]]; then
+            GCP_ENV=stage
+            GCP_PROJECT=moz-fx-productdelivery-no-7d6a
+            FTP_URL_PREFIX="https://ftp.stage.mozaws.net/pub/thunderbird-mobile/android/"
+          fi
+          SERVICE_ACCOUNT="thunderbird-team-${GCP_ENV}@${GCP_PROJECT}.iam.gserviceaccount.com"
+          WORKLOAD_IDENTITY_PROVIDER="projects/324168772199/locations/global/workloadIdentityPools/github-actions/providers/github-actions"
+
+          FTP_SUBDIR="releases"
+          if [[ "${RELEASE_TYPE}" = "daily" ]]; then
+            FTP_SUBDIR="nightly"
+          fi
+
+          echo "SERVICE_ACCOUNT=${SERVICE_ACCOUNT}" >> $GITHUB_OUTPUT
+          echo "WORKLOAD_IDENTITY_PROVIDER=${WORKLOAD_IDENTITY_PROVIDER}" >> $GITHUB_OUTPUT
+          echo "FTP_DESTINATION=${GCP_PROJECT}-productdelivery/pub/thunderbird-mobile/android/${FTP_SUBDIR}/${VERSION_NAME}" >> $GITHUB_OUTPUT
```

Comment:
Overwriting daily was not intended. We could take an approach like desktop does, and have subdirectories per year/month/date. E.g. https://ftp.mozilla.org/pub/thunderbird/nightly/2024/11/2024-11-19-10-02-42-comm-central/

Reply:
I would just go with the following for now:
/pub/thunderbird-mobile/android/nightly/2024/11/2024-11-19-10-48-32-main/

As you mentioned, we can't overwrite via the github action, so I don't think we can push to latest at this point. We will have to revisit that once we move to taskcluster, where we would have:

/pub/thunderbird-mobile/android/nightly/latest-main/

And then releases would stay the same:

/pub/thunderbird-mobile/android/releases/

Reply:
Can we ask the team if they can scope the keys so that an overwrite in /pub/thunderbird-mobile/android/nightly/latest-main/ is possible? This should be low risk, as other daily assets would remain unharmed and the worst thing that could happen is that the latest- file is replaced. I'm not sure it is possible this granular.

Otherwise this looks good. So we'd have the following, did I miss anything?

/pub/thunderbird-mobile/android/nightly/2024/11/2024-11-19-10-48-32-main/thunderbird-10.0a1.apk
/pub/thunderbird-mobile/android/nightly/2024/11/2024-11-19-10-48-32-main/thunderbird-10.0a1.tar.gz
/pub/thunderbird-mobile/android/nightly/2024/11/2024-11-19-10-48-32-main/thunderbird-10.0a1.zip

/pub/thunderbird-mobile/android/releases/9.0/thunderbird-9.0.apk
/pub/thunderbird-mobile/android/releases/9.0/thunderbird-9.0.tar.gz
/pub/thunderbird-mobile/android/releases/9.0/thunderbird-9.0.zip
/pub/thunderbird-mobile/android/releases/9.0/k9mail-9.0.apk
/pub/thunderbird-mobile/android/releases/9.0/k9mail-9.0.tar.gz
/pub/thunderbird-mobile/android/releases/9.0/k9mail-9.0.zip

I'd also be ok only having tar.gz or only zip, not sure it will be useful to have the sources twice.

Reply:
Let's go with just tar.gz